index.js

```diff
@@ -285,17 +285,41 @@ export function escapePath(path) {
 	return path.replace(/[\\.[]/g, '\\$&');
 }
 
+// The keys returned by Object.entries() for arrays are strings
+function entries(value) {
+	if (Array.isArray(value)) {
+		return value.map((value, index) => [index, value]);
+	}
+
+	return Object.entries(value);
+}
+
+function stringifyPath(pathSegments) {
+	let result = '';
+
+	for (let [index, segment] of entries(pathSegments)) {
+		if (typeof segment === 'number') {
+			result += `[${segment}]`;
```

Comment:
This is missing the check for `index === 0`. But wouldn’t it be easier to replace this whole function implementation with

```js
return pathSegments
  .map(segment => typeof segment === 'number' ? `[${segment}]` : escapePath(segment))
  .join('.')

Reply:
> This is missing the check for `index === 0`

No, there is no special behaviour for the first index.

> But wouldn’t it be easier to replace this whole function implementation with

No, indexes should not be seperated by dots.